raster-foundry/raster-foundry
# Filter organizations by visibility and platform

## Overview
- Add visibility column to organizations - default to private
- Respect visibility field of organizations when fetching
- Go to error page when frontend cannot fetch /config correctly
- Respect search query parameter

### Checklist

- ~Styleguide updated, if necessary~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~


### Notes

Users can't view members of public organizations that they are not a part of.
I think handling that, especially given that we're going to be supporting multiple orgs, is something that warrants handling in a separate PR

I updated the dev database since I forgot to add the private org user to the private org, and that's useful for testing this PR out

## Testing Instructions

 * load development data with the updated database `./scripts/load_development_data --download`
 * Log in as `dev@rasterfoundry.com`
 * Go to http://localhost:9091/admin/platform/31277626-968b-4e40-840b-559d9c67863c/organizations and verify that only the RF public organization shows up
* Log in as `systems@rasterfoundry.com` and go to the same page. All orgs on the RF platform should show up. 
  User has permission as a superuser to view all orgs, so if you stick the testing platform id into the url and go there, you should see the public testing org - there's no way to get there through the UI yet.
* Log in as `testing+public+admin@rasterfoundry.com` and `systems+testing@rasterfoundry.com` and verify that you have permission to view the public org of the testing platform but nothing on the RF platform (you may need to move the user to the testing platform and public org for it, not sure if that's in the dev database)
Closes #3451
